src/Command/Push/PushArtifactCommand.php

```diff
@@ -0,0 +1,284 @@
+  }
+
+  /**
+   * @param \Symfony\Component\Console\Input\InputInterface $input
+   * @param \Symfony\Component\Console\Output\OutputInterface $output
+   *
+   * @return int 0 if everything went fine, or an exit code
+   * @throws \Exception
+   */
+  protected function execute(InputInterface $input, OutputInterface $output): int {
+    // @todo change deploy strategy depending on whether this is a "source" repo (requiring building) or not
+    // @todo handle if Git user name/email is missing
+    $this->setDirAndRequireProjectCwd($input);
+    $is_dirty = $this->isLocalGitRepoDirty();
+    $commit_hash = $this->getLocalGitCommitHash();
+    if ($is_dirty) {
+      throw new AcquiaCliException('Pushing code was aborted because your local Git repository has uncommitted changes. Please either commit, reset, or stash your changes via git.');
+    }
+    $this->checklist = new Checklist($output);
+
+    // @todo handle environments with tags deployed
+    $output->writeln('<info>You must select an environment with a Git branch deployed</info>');
+    $environment = $this->determineEnvironment($input, $output, TRUE);
+    if (strpos($environment->vcs->path, 'tags') === 0) {
+      throw new AcquiaCliException("You cannot push to an environment that has a git tag deployed to it. Environment {$environment->name} has {$environment->vcs->path} deployed. Please select a different environment.");
+    }
+    $artifact_dir = Path::join(sys_get_temp_dir(), 'acli-push-artifact');
+    $output_callback = $this->getOutputCallback($output, $this->checklist);
+
+    $this->checklist->addItem('Preparing artifact directory');
+    $this->prepareDir($output_callback, $artifact_dir, $environment->vcs->url, $environment->vcs->path);
+    $this->checklist->completePreviousItem();
+
+    $this->checklist->addItem('Generating build artifact');
+    $this->build($output_callback, $artifact_dir);
+    $this->checklist->completePreviousItem();
+
+    if (!$input->getOption('no-sanitize')) {
+      $this->checklist->addItem('Sanitizing build artifact');
+      $this->sanitize($output_callback, $artifact_dir);
+      $this->checklist->completePreviousItem();
+    }
+
+    $this->checklist->addItem("Committing changes (commit hash: $commit_hash)");
+    $this->commit($output_callback, $artifact_dir, $commit_hash);
+    $this->checklist->completePreviousItem();
+
+    if (!$input->getOption('dry-run')) {
+      $this->checklist->addItem("Pushing changes to <options=bold>{$environment->vcs->path}</> branch in the <options=bold>{$environment->name}</> environment");
+      $this->push($output_callback, $artifact_dir);
+      $this->checklist->completePreviousItem();
+    }
+    else {
+      $this->logger->warning("The <options=bold>--dry-run</> option prevented changes from being pushed to Acquia Cloud. The artifact has been built at <options=bold>$artifact_dir</>");
```

Comment:
Use IO

Reply:
I don't see how to use IO here. When I tried, it wasn't interpreting `<options=bold></>`. In other areas of the code where we have a warning and font styles, we use `$this->logger->warning()`.